solidity/contracts/token/bridge/EverclearTokenBridge.sol

```diff
@@ -0,0 +1,235 @@
+        feeParams = IEverclearAdapter.FeeParams({
+            fee: _fee,
+            deadline: _deadline,
+            sig: _sig
+        });
+        emit FeeParamsUpdated(_fee, _deadline);
+    }
+
+    /**
+     * @notice Sets the output asset address for a destination domain
+     * @dev Only callable by the contract owner
+     * @param _destination The destination domain ID
+     * @param _outputAsset The output asset address on the destination chain (as bytes32)
+     */
+    function setOutputAsset(
+        uint32 _destination,
+        bytes32 _outputAsset
+    ) external onlyOwner {
+        outputAssets[_destination] = _outputAsset;
+        emit OutputAssetSet(_destination, _outputAsset);
+    }
+
+    /**
+     * @notice Sets multiple output assets in a single transaction for gas efficiency
+     * @dev Only callable by the contract owner. Arrays must be the same length
+     * @param _destinations Array of destination domain IDs
+     * @param _outputAssets Array of output asset addresses on destination chains
+     */
+    function setOutputAssetsBatch(
+        uint32[] calldata _destinations,
+        bytes32[] calldata _outputAssets
+    ) external onlyOwner {
+        require(
+            _destinations.length == _outputAssets.length,
+            "ETB: Length mismatch"
+        );
+
+        uint256 len = _destinations.length;
+
+        for (uint256 i = 0; i < len; ) {
+            outputAssets[_destinations[i]] = _outputAssets[i];
+            emit OutputAssetSet(_destinations[i], _outputAssets[i]);
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    /**
+     * @notice Checks if an output asset is configured for a destination
+     * @param _destination The destination domain ID to check
+     * @return True if output asset is set for the destination, false otherwise
+     */
+    function isOutputAssetSet(
+        uint32 _destination
+    ) external view returns (bool) {
+        return outputAssets[_destination] != bytes32(0);
+    }
+
+    /**
+     * @notice Provides a quote for transferring tokens to a remote chain
+     * @dev Returns the gas payment quote and the total token amount needed (including fees)
+     * @param _destination The destination domain ID
+     * @param _recipient The recipient address on the destination chain
+     * @param _amount The amount of tokens to transfer
+     * @return quotes Array of quotes containing gas payment and token amount requirements
+     */
+    function quoteTransferRemote(
+        uint32 _destination,
+        bytes32 _recipient,
+        uint256 _amount
+    ) public view override returns (Quote[] memory quotes) {
+        quotes = new Quote[](2);
+        quotes[0] = Quote({
+            token: address(0),
+            amount: _quoteGasPayment(_destination, _recipient, _amount)
+        });
+
+        quotes[1] = Quote({
+            token: address(wrappedToken),
+            amount: _amount + feeParams.fee
```

Comment:
maybe consider overriding the `_feeAmount` function from my PR and inheriting the `quoteTransferRemote` behavior